src/java.net.http/share/classes/jdk/internal/net/http/common/TimeSource.java

```diff
@@ -0,0 +1,129 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.net.http.common;
+
+import java.time.Instant;
+import java.time.InstantSource;
+
+/**
+ * An {@link InstantSource} based on {@link System#nanoTime()} for the
+ * purpose of handling timeouts. This time source is intentionally not
+ * based on the system clock, in order to not be sensitive to clock skews
+ * caused by changes to the wall clock. Consequently, callers should use
+ * instants returned by this time source solely for the purpose of
+ * comparing them with other instants returned by this same time source.
+ * The granularity of instants returned is identical to the granularity
+ * of {@link System#nanoTime()}. This time source has the same property
+ * of monotonicity than the {@link System#nanoTime()} it is based on.
+ */
+public final class TimeSource implements InstantSource {
+    private static volatile NanoSource nanoSource = new NanoSource();
+    private NanoSource localSource = nanoSource;
+    private static final TimeSource SOURCE = new TimeSource();
+
+    private static final class NanoSource implements InstantSource {
+        static final int TIME_WINDOW = Integer.MAX_VALUE;
+        final Instant first;
+        final long firstNanos;
+        NanoSource() {
+            this(Instant.now(), System.nanoTime());
+        }
+        NanoSource(Instant first, long firstNanos) {
+            this.first = first;
+            this.firstNanos = firstNanos;
+        }
+
+        @Override
+        public Instant instant() {
+            return instant(System.nanoTime());
+        }
+
+        Instant instant(long nanos) {
+            return instant(nanos, nanos - firstNanos);
+        }
+
+        Instant instant(long nanos, long delay) {
+            Instant now = first.plusNanos(delay);
+            if (!isInWindow(delay)) {
+                // Shifts the time reference (firstNanos) to
+                // prevent issues that may be caused by
+                // System.nanoTime() - firstNanos wrapping
+                // around.
+                // The use of Integer.MAX_VALUE is arbitrary.
+                // Any value not too close to Long.MAX_VALUE
+                // would do.
+                nanoSource = new NanoSource(now, nanos);
+            }
+            return now;
+        }
+
+        // @ForceInline
```

Comment:
Should these couple of commented out `@ForceInline` be deleted?

Reply:
Done